### PR TITLE
Fix Host step

### PR DESCRIPTION
**Pull request by VDubber**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -191,19 +192,55 @@ export class APIClient {
...
+          },
+        },
+      );
+
+      const data = await response.json();
+      for (const host of data.host_list) {
+        await iteratee(host);
+      }
+
+      cursor = data.totalReturned + 1;
```
Comment (adam-in-ict) [inline src/client.ts:240]: Should this be `data.total_returned` like the line below?

Comment (VDubber) [reply src/client.ts:240]: Yes!

File: src/client.ts [hunk 2]
```diff
@@ -191,19 +192,55 @@ export class APIClient {
...
+    const requestSize = 1000;
+    let cursor = 0;
+    let totalReturned = 0;
+    let totalMatching = 0;
+
+    do {
+      const url = `https://api.${this.config.datadogHost}/api${endpoint}?count=${requestSize}&start=${cursor}`;
+
+      const response = await retry(
+        async () => {
```
Comment (eXtremeX) [inline src/client.ts:212]: The `datadog_host` env field is optional. If it's blank, we'll need to add some code to point it default value of `datadoghq.com` like we do in the constructor.

Perhaps something like this:
```suggestion
   const host = this.config.datadogHost ? this.config.datadogHost : 'datadoghq.com';

    do {
      const url = `https://api.${host}/api${endpoint}?count=${requestSize}&start=${cursor}`;

      const response = await retry(
        async () => {
```

File: src/client.ts (hunk 1 above)
Comment (VDubber) [reply src/client.ts:240]: I'll install data dog on my other machine so I can test this.

Comment (eXtremeX) [reply src/client.ts:240]: Yep, this seems to be the case, just verifying with our account.

image

**Comment by VDubber** [comment]
DataDog updated their npm package with a fix. A followup PR will replace this.

**Pull request closed without merging**